server/server.go

```diff
@@ -1,23 +1,53 @@
 package server
 
 import (
-	"net/http"
-	"net"
+	"context"
 	"fmt"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"os/exec"
+	"strings"
 	"sync"
+	"sync/atomic"
 	"time"
-	"github.com/gruntwork-io/health-checker/options"
+
 	"github.com/gruntwork-io/gruntwork-cli/errors"
+	"github.com/gruntwork-io/health-checker/options"
 )
 
+const DEFAULT_CHECK_TIMEOUT = 5
+
+type TcpCheck struct {
+	Name string `yaml:"name"`
+	Host string `yaml:"host"`
+	Port int `yaml:"port"`
+	Timeout int `yaml:"timeout"`
+}
+
+type HttpCheck struct {
+	Name string `yaml:"name"`
+	Host string `yaml:"host"`
+	Port int `yaml:"port"`
+	SuccessStatusCodes []int `yaml:"success_status_codes"`
+	BodyRegex string `yaml:"body_regex"`
+	Timeout int `yaml:"timeout"`
```

Comment:
`time.Duration` would be a more appropriate type here, but if the yaml parser can't handle that, then just ignore me and consider calling the field `TimeoutSeconds`.